main.tf

```diff
@@ -96,26 +91,26 @@ resource "azurerm_network_security_rule" "outbound" {
 }
 
 ##-----------------------------------------------------------------------------
-## Below resource will associate above created network security group to subnet. 
+## Below resource will associate above created network security group to subnet.
 ##-----------------------------------------------------------------------------
 resource "azurerm_subnet_network_security_group_association" "example" {
   count                     = var.enabled ? length(var.subnet_ids) : 0
   subnet_id                 = element(var.subnet_ids, count.index)
-  network_security_group_id = join("", azurerm_network_security_group.nsg.*.id)
+  network_security_group_id = join("", azurerm_network_security_group.nsg[*].id)
```

Comment:
Pls remove join and use [0] instead of [*]

Reply:
fixed